pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,232 @@
+		),
+		Long: fmt.Sprintf(""+
+			"This program returns the amount of free disk space (in bytes) for a given storage class or compares if there is enough space to hold a certain\n"+
+			"amount of data (--bigger-than). For OpenEBS, when no --bigger-than flag is provided, this program returns a list of nodes and their respective\n"+
+			"free space, while for Rook storage only the available space is returned. When --bigger-than flag is used this program sets the exit code to zero\n"+
+			"if the space available is bigger than the quantity provided. For OpenEBS, if no node has been provided through --openebs-node-name the exit code\n"+
+			"will be zero only if all nodes free space are bigger than the provided quantity (see Examples section for more details).\n\n"+
+			"Supports the following storage provisioners: %s, %s, %s",
+			openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner,
+		),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			k8sConfig, err := config.GetConfig()
+			if err != nil {
+				return fmt.Errorf("failed to read kubernetes configuration: %w", err)
+			}
+
+			clientSet, err = kubernetes.NewForConfig(k8sConfig)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %w", err)
+			}
+
+			rookClientSet, err = rookcli.NewForConfig(k8sConfig)
+			if err != nil {
+				return fmt.Errorf("failed to create rook client: %w", err)
+			}
+
+			if selectedClass, err = getStorageClassByName(cmd.Context(), clientSet, forStorageClass); err != nil {
+				return err
+			}
+
+			if biggerThanString == "" {
+				return nil
+			}
+
+			parsed, err := resource.ParseQuantity(biggerThanString)
+			if err != nil {
+				return fmt.Errorf("failed to parse %s as a quantity: %w", biggerThanString, err)
+			}
+			biggerThanBytes = parsed.Value()
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancel := signal.NotifyContext(cmd.Context(), syscall.SIGTERM, syscall.SIGINT)
+			defer cancel()
+
+			switch selectedClass.Provisioner {
+			case openEBSLocalProvisioner:
+				return evaluateOpenEBSFreeSpace(ctx, clientSet, openEBSImage, selectedClass.Name, openEBSNode, biggerThanBytes)
+
+			case rookCephFSProvisioner, rookRBDProvisioner:
+				return evaluateRookFreeSpace(ctx, clientSet, rookClientSet, selectedClass.Name, biggerThanBytes)
+
+			default:
+				fmt.Printf("Provisioner %q is not supported, unable to determine free space.\n", selectedClass.Provisioner)
+				return nil
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&forStorageClass, "storageclass", "", "Inform the storage class name for which to check the free disk space. If not informed the default storage will be used.")
+	cmd.Flags().StringVar(&biggerThanString, "bigger-than", "", "Compares if the cluster free disk space is bigger than the provided value. Accepts the same format as used when defining storage requests in Kubernetes (e.g. 10G, 5Gi, 500M).")
```

Comment:
```suggestion
```
Why do we need this one?
Do we really a command that checks if the value is > than an a value ? 
Can not this check be done where this info is useful?